core/vm/contracts.go

```diff
@@ -1135,3 +1158,37 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 
 	return h
 }
+
+// P256VERIFY (secp256r1 signature verification)
+// implemented as a native contract
+type p256Verify struct{}
+
+// RequiredGas returns the gas required to execute the precompiled contract
+func (c *p256Verify) RequiredGas(input []byte) uint64 {
+	return params.P256VerifyGas
+}
+
+// Run executes the precompiled contract with given 160 bytes of param, returning the output and the used gas
+func (c *p256Verify) Run(input []byte) ([]byte, error) {
+	// Required input length is 160 bytes
+	const p256VerifyInputLength = 160
+	// Check the input length
+	if len(input) != p256VerifyInputLength {
+		// Input length is invalid
+		return nil, nil
+	}
+
+	// Extract the hash, r, s, x, y from the input
+	hash := input[0:32]
+	r, s := new(big.Int).SetBytes(input[32:64]), new(big.Int).SetBytes(input[64:96])
+	x, y := new(big.Int).SetBytes(input[96:128]), new(big.Int).SetBytes(input[128:160])
+
+	// Verify the secp256r1 signature
+	if secp256r1.Verify(hash, r, s, x, y) {
+		// Signature is valid
+		return common.LeftPadBytes(common.Big1.Bytes(), 32), nil
+	} else {
+		// Signature is invalid
+		return nil, nil
```

Comment:
AFAICT not returning a value here, results [in the return value of the call being set to nil](https://github.com/ethereum-optimism/op-geth/blob/a5a4fa7032bb248f5a7c40f4e8df2b131c4186a4/core/vm/instructions.go#L684). 

However it does match the [spec](https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md#precompiled-contract-specification) so likely isn't an implementation issue.

FWIW some solidity implementations([example](https://github.com/daimo-eth/p256-verifier/blob/master/src/P256Verifier.sol#L11-L39)) do set it to 0, though I don't think solidity has a nil value -- so that maybe not by choice.

@ulerdogan I'm curious if there's any advantages to using nil vs. zero bytes?

Reply:
Good discussion! I don't think it has a significant advantage or disadvantage for both cases. As the proposal is Final, I don't recommend changing its behavior.
The choice of returning `nil` as return in the spec has been shaped by the feedbacks I received while we were preparing the EIP, now similar to `ecrecover` precompile. So, this implementation is compatible with RIP-7212 spec.

Reply:
Thanks for the background info, I also didn't mean to propose to change the implementation, as the spec is already final. I just wanted to understand the reasoning behind it. It appears to me that it would have been easier to work with if it was always guaranteed to return a 0 or 1 uint256.